src/core/net/ip6.cpp

```diff
@@ -1492,6 +1505,41 @@ bool Ip6::IsOnLink(const Address &aAddress) const
     return rval;
 }
 
+#if OPENTHREAD_CONFIG_IP6_BR_COUNTERS_ENABLE
+void Ip6::UpdateBorderRoutingCounters(const Message &    aMessage,
+                                      const Header &     aHeader,
+                                      otPacketsAndBytes &aUnicastCounter,
+                                      otPacketsAndBytes &aMulticastCounter)
+{
+    otPacketsAndBytes *counter = nullptr;
+
+    if (!Get<Netif>().GetUnicastAddresses().ContainsMatching(aHeader.GetSource()) &&
+        !Get<Netif>().GetUnicastAddresses().ContainsMatching(aHeader.GetDestination()) &&
```

Comment:
Can we use `Get<Netif>().HasUnicastAddress()` instead of 
`Get<Netif>().GetUnicastAddress().ContainsMatching()`?